scripts/precompile-headers.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+BIN=$(dirname "$0")/
+HILTI_CONFIG=${BIN}/hilti-config
+
+for flag in $(${HILTI_CONFIG} --cxxflags); do
+    if ! echo "${flag}" | grep -q '^-I'; then
+        continue
+    fi
+    dir=${flag#??}
+    if [ -e "${dir}"/hilti/rt/libhilti.h ]; then
+        LIBHILTI=${dir}/hilti/rt/libhilti.h
+    fi
+done
+
+if [ -z "${var+x}" ]; then
+    echo "Error: could not determine location of libhilti.h"
+    exit 1
+fi
+
+VERSION=$(${HILTI_CONFIG} --version | cut -d ' ' -f1)
+
+CACHE=$HOME/.cache/spicy/${VERSION}
+echo "Clearing cache directory $CACHE"
+rm -rf "${CACHE}"
+mkdir -p "${CACHE}"
+
+cp "${LIBHILTI}" "${CACHE}/precompiled_libhilti_debug.h"
+echo "Creating ${CACHE}/precompiled_libhilti_debug.h.pch"
+$("${HILTI_CONFIG}" --cxx --cxxflags --debug) -x c++-header "${LIBHILTI}" -o "${CACHE}/precompiled_libhilti_debug.h.pch"
```

Comment:
Just checking: this works for both GCC and clang the same way?

Reply:
I checked again, and yes, this script works for GCC as well.

There is however still an issue in how with this patch we by default enable JIT, at least for developers. The precompiled headers will (in `hilti/runtime/CMakeLists.txt`) be created with the compiler configured for CMake. JIT however always compiles through clang (in `compileModule`). This means that if spicy was compiled with GCC, one cannot use precompiled headers together with JIT as clang cannot consume GCC precompiled headers. A similar problem exists if e.g., the llvm root used (`./configure`'s `--with-llvm-root`) is older than the clang compiler used to compile Spicy as clang PCHs seem mostly incompatible between versions.

Is making this behavior optional, but by default enabled a good enough solution?

Reply:
Yeah, seems ok if PCHs aren't supported if one mixes compilers some way.